Custom Sound Example in Wiki. (#4179)
GregTechCEu/GregTechCEu-1.19: docs/content/Modpacks/Other-Topics/Custom-Sounds.md

Reviewer: You should also describe how to actually get the data to generate properly as it's not obvious

---
title: Custom Sounds
---

## Creating a custom sound

!!! Warning

>     Registering custom sounds is currently only supported in Java. Although you can call your sound in kubejs once defined.

Reviewer: ```suggestion
    Registering custom sounds is currently only supported in Java, though you can use your sound in kubejs scripts once it's defined.
```

> To add a new sound, a sounds class is required. 

Reviewer: ###### well, not exactly... It's just recommended to put them in a separate class

This class prepares for registrate to register the sounds. 
An example of a custom sound can be found below.

```java

> import static com.gregtechceu.gtceu.common.registry.GTRegistration.REGISTRATE;

Reviewer: Do. Not. Use. GT's registrate.
Defining sounds requires a bit of datagen which **will not run** for your mod if you use GT's registrate instance.
It'd also register them to the wrong nsmespace.

public class ExampleSound {

>     public static final SoundEntry MICROVERSE = REGISTRATE.sound(examplemod.id("microverse")).build();

Reviewer: ```suggestion
    public static final SoundEntry MICROVERSE = REGISTRATE.sound("microverse").build();
```

    public static void init() {}
}
```

> After you make this class and init it inside your main mod class, you want to setup datagen for the sound.

Reviewer: ```suggestion
After you make this class and initialize it in your main mod class, you want to setup datagen for the sounds.
It's a bit more complicated than normal datagen, so an example can be found below.
```
If I didn't manage to select the 2nd line (am on mobile) please do the change manually

It's a bit more complicated than normal datagen so an example can be found below.

```java

> @Mod.EventBusSubscriber(modid = examplemod.MOD_ID, bus = Mod.EventBusSubscriber.Bus.MOD)

Reviewer: ```suggestion
@Mod.EventBusSubscriber(modid = ExampleMod.MOD_ID, bus = Mod.EventBusSubscriber.Bus.MOD)
```

public class ExampleDataGenerators {

    @SubscribeEvent
    public static void gatherData(GatherDataEvent event) {
        PackOutput packOutput = event.getGenerator().getPackOutput();

        if (event.includeClient()) {
            event.getGenerator().addProvider(
                    true,
                    new SoundEntryBuilder.SoundEntryProvider(packOutput, examplemod.MOD_ID));
        }
    }
}

```

> Finally, you need to actually prepare your sound for use. For a sound to be registered it must be in .ogg format and be inside assets/examplemod/sounds. 

Reviewer: This has to be done before running datagen or it'll fail

Author: Redid it a bit, lemme know if it still needs some refining.

> Mono sounds are localized to the area around the machine, while stereo sounds are global.

Reviewer: ```suggestion
!!! note "mono vs. stereo audio"

    Your audio file should be mono, as Minecraft's attentuation logic only works with single-channel audio. Stereo sounds won't fade out (they'll be played at the same volume at any distance from the source) and should only be used for background tracks such as the main menu music. 
```

---
title: Custom Sounds
---

> ## Creating a custom sound

Reviewer 2: ```suggestion
## Creating a Custom Sound
```

!!! Warning
    Registering custom sounds is currently only supported in Java, though you can use your sound in kubejs scripts once it's defined.

To add a new sound, a sounds class is required. 
This class prepares for registrate to register the sounds. 
An example of a custom sound can be found below.

```java
import static com.examplemod.common.registry.ExampleRegistration.REGISTRATE;

public class ExampleSound {

    public static final SoundEntry MICROVERSE = REGISTRATE.sound("microverse").build();

    public static void init() {}
}
```

> Before you run datagen, the sound needs to be prepared for use. For a sound to be registered it must be in .ogg format and be inside assets/examplemod/sounds. 

Reviewer 2: ```suggestion
Before you run datagen, the sound needs to be prepared for use. For a sound to be registered it must be in .ogg format and be inside `assets/examplemod/sounds`. 
```
